core/templates/domain/admin/admin-backend-api.service.ts

```diff
@@ -140,8 +140,70 @@ export class AdminBackendApiService {
       });
     });
   }
-}
+  flushCache():Promise<void> {
+    return this.http.post<void>(
+      AdminPageConstants.ADMIN_MEMORY_CACHE_HANDLER_URL, {}
+    ).toPromise();
+  }
+
+  clearSearchIndex():Promise<void> {
+    return this.http.post<void>(
+      AdminPageConstants.ADMIN_HANDLER_URL, {}
+    ).toPromise();
+  }
+
+  regenerateTopicRelatedOpportunities(topicId:string):Promise<void> {
+    return this.http.post<void>(
+      AdminPageConstants.ADMIN_HANDLER_URL, {
+        action: 'regenerate_topic_related_opportunities',
+        topic_id: topicId
+      }
+    ).toPromise();
+  }
+
+  uploadTopicSimilarities(data:string | ArrayBuffer):Promise<void> {
```

Comment:
Why the type is ArrayBuffer? Use an interface to define the dict structure if needed.

Reply:
done!